core/trino-main/src/main/java/io/trino/operator/output/TaskOutputOperator.java

```diff
@@ -153,8 +166,10 @@ public void addInput(Page page)
         }
 
         page = pagePreprocessor.apply(page);
-
         outputBuffer.enqueue(splitAndSerializePage(page));
+        long lastBufferUtilizationRecorded = ticker.read();
+        outputBufferUtilization.add(outputBuffer.getUtilization(), (double) (lastBufferUtilizationRecorded - this.lastBufferUtilizationRecorded));
```

Comment:
Histogram should be tracked by `OutputBufferMemoryManager#updateMemoryUsage`. Otherwise you don't account for pages being removed from buffer